daemon/algod/api/server/v1/handlers/handlers.go

```diff
@@ -163,16 +164,33 @@ func participationKeysEncode(r basics.AccountData) *v1.Participation {
 	return &apiParticipation
 }
 
+// sanitizePrintableUTF8String checks to see if the entire string is a UTF8 printable string.
+// If this is the case, the string is returned as is. Otherwise, the empty string is returned.
+func sanitizePrintableUTF8String(in string) string {
+	// check to see if all the characters in the input string are utf-8.
+	if !utf8.ValidString(in) {
+		return ""
+	}
+	// iterate throughout all the characters in the string to see if they are all printable.
+	for _, c := range in {
+		// is this a printable character ?
+		if !unicode.IsPrint(c) {
+			return ""
+		}
+	}
+	return in
+}
+
 func modelAssetParams(creator basics.Address, params basics.AssetParams) v1.AssetParams {
 	paramsModel := v1.AssetParams{
 		Total:         params.Total,
 		DefaultFrozen: params.DefaultFrozen,
 		Decimals:      params.Decimals,
 	}
 
-	paramsModel.UnitName = strings.TrimRight(string(params.UnitName[:]), "\x00")
-	paramsModel.AssetName = strings.TrimRight(string(params.AssetName[:]), "\x00")
-	paramsModel.URL = strings.TrimRight(string(params.URL[:]), "\x00")
+	paramsModel.UnitName = sanitizePrintableUTF8String(params.UnitName)
```

Comment:
Is this not backwards compatible? What used to be some value will be empty now?

Reply:
it does break backward compatibility - but only for the cases of non uff-8; currently, I'm unaware of any such cases.